src/odemis/gui/cont/project.py

```diff
@@ -230,8 +235,9 @@ def remove_roa_ctrl(self, roa_ctrl):
         roa_ctrl.panel.Destroy()
         self.panel.fit_panels()
 
-        # Remove controller from .roa_ctrl list
-        self.roa_ctrls = {key: val for key, val in self.roa_ctrls.items() if val != roa_ctrl}
+        # Remove roa controller and its subscriber callback
+        self.roa_ctrls.pop(roa_ctrl)
```

Comment:
If you don't care about the value associated, and are certain the key is present, you can just write `del self.roa_ctrls[roa_ctrl]`.
If you're not certain it is still present, then you should do `.pop(..., None)`.